pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConsumer.java

```diff
@@ -69,8 +69,12 @@ public synchronized MessageBatch<StreamMessage<byte[]>> fetchMessages(long start
     ConsumerRecords<String, Bytes> consumerRecords = _consumer.poll(Duration.ofMillis(timeoutMillis));
     List<ConsumerRecord<String, Bytes>> messageAndOffsets = consumerRecords.records(_topicPartition);
     List<StreamMessage<byte[]>> filtered = new ArrayList<>(messageAndOffsets.size());
+    long firstOffset = startOffset;
     long lastOffset = startOffset;
     StreamMessageMetadata rowMetadata = null;
+    if (!consumerRecords.isEmpty()) {
+      firstOffset = consumerRecords.iterator().next().offset();
+    }
```

Comment:
Should we get offset from records under the topic partition?
```suggestion
    if (!messageAndOffsets.isEmpty()) {
      firstOffset = messageAndOffsets.get(0).offset();
    }
```

Reply:
There is also a check for `offset >= startOffset` underlying. Is it even possible?

Reply:
Yes. that condition is checking if the offset of a message is between start, end offset while it is looping through the messages in the batch.

Reply:
regarding the suggestion, it is better to check the offset of the first message. Lets say in a batch only the last message is from a topic. Then this will give the wrong signal that the offset jumped.